src/rules/fileNameCasingRule.ts

```diff
@@ -25,47 +25,95 @@ enum Casing {
+            One argument which is either a string defining the file casing or an array consisting of a file name
+            matches and corresponding casing strings.
 
-            * \`${Casing.CamelCase}\`: File names must be camel-cased: \`fileName.ts\`.
-            * \`${Casing.PascalCase}\`: File names must be Pascal-cased: \`FileName.ts\`.
-            * \`${Casing.KebabCase}\`: File names must be kebab-cased: \`file-name.ts\`.
-            * \`${Casing.SnakeCase}\`: File names must be snake-cased: \`file_name.ts\`.`,
+            * In both cases the casing string must be one of the options:
+            ** \`${Casing.CamelCase}\`: File names must be camel-cased: \`fileName.ts\`.
+            ** \`${Casing.PascalCase}\`: File names must be pascal-cased: \`FileName.ts\`.
+            ** \`${Casing.KebabCase}\`: File names must be kebab-cased: \`file-name.ts\`.
+            ** \`${Casing.SnakeCase}\`: File names must be snake-cased: \`file_name.ts\`.
+        
+            * The array again consists of array with two items. The first item must be a case-insenstive
+            regular expression to match files, the second item must be a valid casing option (see above)`,
         options: {
-            type: "array",
-            items: [
-                {
-                    type: "string",
-                    enum: [Casing.CamelCase, Casing.PascalCase, Casing.KebabCase, Casing.SnakeCase],
-                },
-            ],
+            type: "list",
+            listType: {
+                anyOf: [
+                    {
+                        type: "string",
+                        enum: [
+                            Casing.CamelCase,
+                            Casing.PascalCase,
+                            Casing.KebabCase,
+                            Casing.SnakeCase
+                        ]
+                    },
+                    {
+                        type: "array",
+                        items: [
+                            {
+                                type: "string"
+                            },
+                            {
+                                type: "string",
+                                enum: [
+                                    Casing.CamelCase,
+                                    Casing.PascalCase,
+                                    Casing.KebabCase,
+                                    Casing.SnakeCase
+                                ]
+                            }
+                        ]
+                    }
+                ]
+            }
         },
         optionExamples: [
             [true, Casing.CamelCase],
             [true, Casing.PascalCase],
             [true, Casing.KebabCase],
             [true, Casing.SnakeCase],
+            [
+                true,
+                [
+                    [".style.ts$", Casing.KebabCase],
+                    [".tsx$", Casing.PascalCase],
+                    [".*", Casing.CamelCase]
```

Comment:
It's generally convention to use objects for key/value pairs. Unless you have a specific need, please switch to something like:

```typescript
{
    ".style.ts$": Casing.KebabCase,
    ".tsx$": Casing.PascalCase,
    ".*": Casing.CamelCase,
}
```

One benefit, in addition to being semantically more clear, is that you get rid of duplicate keys easily.

Reply:
Will switch to object, thanks!
I'm embarassed to admit how long it took me to realize that the syntax in `options` is actually JSON schema (since I never really worked with that). After that I optimized the declaration a bit, looking forward to your opinion after I pushed my changes.
Maybe the comment for `options` in `IRuleMetadata` should begin with "_JSON_ schema ..." for people like me 😄